api/test/wfl/integration/datarepo_test.clj

```diff
@@ -59,3 +61,28 @@
                  (datarepo/ingest-table dataset-id table-url table-name))]
             (is (= 1 row_count))
             (is (= 0 bad_row_count))))))))
+
+(deftest test-create-snapshot
+  ;; To test given a dataset, we can query BigQuery for row-ids
+  ;; and use them to create a snapshot!
+  (let [tdr-profile (env/getenv "WFL_TDR_DEFAULT_PROFILE")
+        definition "sarscov2-illumina-full-inputs.json"]
+    (testing "creating snapshots from dataset"
+      (fixtures/with-temporary-dataset
+        (datasets/unique-dataset-request tdr-profile definition)
+        (fn [dataset-id]
+          (let [{:keys [dataProject] :as dataset} (datarepo/dataset dataset-id)
+                table     "sarscov2_illumina_full_inputs"
+                today     (util/datetime->str (util/now))
+                yesterday (util/datetime->str (util/n-day-from-now -1))
+                row-ids   (->> (datarepo/compose-snapshot-query dataset table yesterday today)
+                               (bigquery/query-sync dataProject)
+                               (bigquery/flatten-rows)
+                               ; the fixture does not support cleaning up multiple snapshots yet
+                               (take 1))]
```

Comment:
Can't you snapshot all these rows at once? Why do you need multiple snapshots?

Reply:
Hmm, I think this cline and comment might be outdated, at that moment I wasn't sure how many rows I would use (and there's a hard limit requires it to split at 500 batches), that piece was moved to the other module. I will update this.